src/config/options.ts

```diff
@@ -30,6 +30,8 @@ export type AppsignalOptions = {
   runningInContainer: boolean
   sendParams: boolean
   skipSessionData: boolean
+  testMode: boolean
+  testModeFilePath: string
```

Comment:
Other test specific config we've only added as env vars, like `APPSIGNAL_DIAGNOSE_ENDPOINT` (should have a `_` prefix) for pointing to the local endpoint in the test suite. I don't think this needs to be part of the config class because it doesn't need to be configurable by our users. You can use a `_APPSIGNAL_TEST_MODE` and `_APPSIGNAL_TEST_MODE_FILE_PATH` env var for this.

Reply:
If it's not declared on the config type, then TS won't let you access it from `this.config.data`. I don't think our config object currently supports having configuration options that can't be set via the constructor.

Regarding the `_` prefix at the beginning of the environment variable, I thought that was used only for variables that are passed on to the agent.

We could, of course, just read `process.env["APPSIGNAL_TEST_MODE"]` where needed, instead of getting it from the config object, if that's preferable.

Reply:
> Regarding the _ prefix at the beginning of the environment variable, I thought that was used only for variables that are passed on to the agent.

It's for private env vars, which include the extension's env vars.

> We could, of course, just read process.env["APPSIGNAL_TEST_MODE"] where needed, instead of getting it from the config object, if that's preferable.

I would prefer it, just because it's not a config option any of our users need to be able to set. (That they can potentially via the env var is a minimal risk.)